### PR TITLE
Implement SetAutoPromotionForStage API

### PR DESCRIPTION
**Pull request by devholic**
https://github.com/akuity/kargo/pull/489

Simplify PromotionPolicy management API by adding field-specific set API.

### CODE REVIEW

File: api/service/v1alpha1/service.proto [hunk 1]
```diff
@@ -13,19 +13,26 @@ service KargoService {
...
   rpc ListStages(ListStagesRequest) returns (ListStagesResponse);
   rpc GetStage(GetStageRequest) returns (GetStageResponse);
   rpc UpdateStage(UpdateStageRequest) returns (UpdateStageResponse);
   rpc DeleteStage(DeleteStageRequest) returns (DeleteStageResponse);
   rpc PromoteStage(PromoteStageRequest) returns (PromoteStageResponse);
 
+  /* PromotionPolicy APIs */
+
+  rpc SetAutoPromotionForStage(SetAutoPromotionForStageRequest) returns (SetAutoPromotionForStageResponse);
   rpc CreatePromotionPolicy(CreatePromotionPolicyRequest) returns (CreatePromotionPolicyResponse);
```
Comment (gdsoumya) [inline api/service/v1alpha1/service.proto:28]: I think these 2 should be the same api, don't see any difference between creating a new promotion policy or setting auto enable for a stage. If the promotion policy already exists Update can be used to toggle the auto field imo. In short the backend should just implement simple crud and the clients (cli, ui) should implement wrappers for ux.

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: api/service/v1alpha1/service.proto (hunk 1 above)
Comment (gdsoumya) [reply api/service/v1alpha1/service.proto:28]: Hmm, but I don't see any benefit of having a separate rpc that needs to be maintained when we can do the same with existing rpcs. Also just using the crud api gives the user interfaces way more control over what they want to customize which usually means fewer backend/api changes when minor modifications are needed in the future. I will let @krancour and you decide but I personally feel having separate api for this case doesn't provide any benefits

Comment (krancour) [reply api/service/v1alpha1/service.proto:28]: Yes... they're redundant... but imho, it is the CRUD ops on PromotionPolicies that should go away and be replaced with ops that simply enable/disable autopromotions.

The idea is to hide complexity (like the fact that multiple promotion policies for a single stage create ambiguity) from clients.

Comment (gdsoumya) [reply api/service/v1alpha1/service.proto:28]: If we are going to remove the crud ops and just have specific rpcs like enable/disable then that's fine too. We preferably don't want to maintain duplicates for the same set of actions.

**Review by gdsoumya — APPROVED** [review]

**Pull request merged**